core/src/main/java/com/linecorp/armeria/client/endpoint/StickyEndpointSelectionStrategy.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.endpoint;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.function.ToLongFunction;
+
+import com.google.common.hash.Hashing;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.Endpoint;
+
+/**
+ * An {@code EndpointSelector} strategy which implements sticky load-balancing using
+ * user passed {@code ToLongFunction<CLientRequestContext>} as hash calculator for
+ * consistent hashing.
+ */
+public final class StickyEndpointSelectionStrategy implements EndpointSelectionStrategy {
+
+    private final ToLongFunction<ClientRequestContext> requestHasher;
+
+    /**
+     * Creates a new {@link StickyEndpointSelectionStrategy}
+     * with default hash function to hash a {@link ClientRequestContext} to a {@code long}.
+     *
+     * @param requestHasher The default {@link ToLongFunction} of {@link ClientRequestContext}
+     */
+    public StickyEndpointSelectionStrategy(ToLongFunction<ClientRequestContext> requestHasher) {
+        this.requestHasher = requireNonNull(requestHasher);
+    }
+
+    /**
+     * Creates a new {@link StickyEndpointSelector}.
+     *
+     * @param endpointGroup an {@link EndpointGroup}
+     * @return a new {@link StickyEndpointSelector}
+     */
+    @Override
+    @SuppressWarnings("unchecked")
```

Comment:
Just curious, do we need this `SuppressWarnings`?

Reply:
I don't know, just maintain consistency with another file, may be it's left from previous implementation where we EndpoinSelector has type parameter?
@trustin may know the rationale